packages/design-system/src/Avatar/Avatar.tsx

```diff
@@ -0,0 +1,137 @@
+import React, { ForwardedRef } from "react";
+import clsx from "classnames";
+
+import {
+  AvatarGroupAvatarProps,
+  AvatarGroupProps,
+  AvatarProps,
+} from "./Avatar.types";
+import {
+  AvatarMenuItem,
+  StyledAvatar,
+  StyledAvatarGroup,
+} from "./Avatar.styles";
+import { Icon } from "Icon";
+import {
+  AvatarClassName,
+  AvatarImageClassName,
+  AvatarSvgClassName,
+  AvatarLetterClassName,
+  AvatarGroupClassName,
+} from "./Avatar.constants";
+import { Tooltip } from "Tooltip";
+import { Menu, MenuItem, MenuTrigger } from "Menu";
+import { MenuContent } from "Menu";
+
+const Avatar = React.forwardRef(
+  (props: AvatarProps, ref: ForwardedRef<HTMLSpanElement>) => {
+    const {
+      className,
+      firstLetter,
+      image,
+      label,
+      size = "md",
+      svgIconName = "user-3-line",
+      tooltipPlacement = "bottom",
+      ...rest
+    } = props;
+    return (
+      <Tooltip content={label} placement={tooltipPlacement}>
+        <StyledAvatar
+          aria-label={label}
+          className={clsx(AvatarClassName, className)}
+          size={size}
+          {...rest}
+          ref={ref}
+        >
+          {image && (
+            <img alt="avatar" className={AvatarImageClassName} src={image} />
+          )}
+          {!image && firstLetter && (
+            <span className={AvatarSvgClassName}>{firstLetter}</span>
+          )}
+          {!image && !firstLetter && (
+            <Icon
+              className={AvatarLetterClassName}
+              name={svgIconName}
+              size={size}
+            />
+          )}
+        </StyledAvatar>
+      </Tooltip>
+    );
+  },
+);
+
+Avatar.displayName = "Avatar";
+
+Avatar.defaultProps = {};
+
+function AvatarGroup(props: AvatarGroupProps) {
+  const {
+    avatars,
+    className,
+    maxAvatars = 3,
+    size = "md",
+    tooltipPlacement = "bottom",
+  } = props;
+  const mainAvatars = avatars.slice(0, maxAvatars);
+  const restAvatars = avatars.slice(maxAvatars);
+  return (
+    <StyledAvatarGroup className={clsx(AvatarGroupClassName, className)}>
+      {mainAvatars.map((avatar: AvatarGroupAvatarProps, index: number) => {
+        return (
+          <Avatar
+            className={avatar.className}
+            firstLetter={avatar.firstLetter}
+            image={avatar.image}
+            key={index.toString()}
+            label={avatar.label}
+            size={size}
+            svgIconName={avatar.svgIconName}
+            tooltipPlacement={tooltipPlacement}
+          />
+        );
+      })}
+      {restAvatars.length > 1 && (
```

Comment:
<img width="213" alt="Screenshot 2023-03-24 at 11 44 23 AM" src="https://user-images.githubusercontent.com/13763558/227440401-a622d834-6516-4a4e-9f4f-0e0323c029f9.png">
This is such a strange interaction. Can we not show this tooltip for the "+9" avatar and instead open the menu on hover?

Reply:
This needs a little rework in current structure. Will take up in next phase.